Add vis-skip-repeating-val config enabled by default

disconnect3d opened the pull request:
1. Adds a `vis-skip-repeating-val` config which is enabled by default.
2. Adds a `--no-skip` flag to `vis` command that disables the skipping of repeating values
3. Fixes a bug in `vis` where an empty line was printed at start
4. Changes `_` to `-` in other options/flags of `vis` command

This config changes the `vis` command rendering so that it collapses a heap chunk display like this, instead of repeating its lines:
```
  0x555555559000    0x0000000000000000    0x0000000000000291    ................
  0x555555559010    0x0000000000000000    0x0000000000000000    ................
      ... ↓      39 repeated lines skipped
  0x555555559290    0x0000000000000000    0x0000000000000111    ................
```

<!--
    Please make sure to read the linting and testing instructions at
    https://pwndbg.re/dev/contributing/ before creating a PR.
-->

+ [x] I read the contributing documentation.
+ [x] I am providing a screenshot of the (new feature) / (fixed bug).

See screenshot in the post below (it includes the empty line which was removed, but it shows the skipping of repeated values).

disconnect3d committed:
Add vis-skip-repeating-val config enabled by default

Adds a `vis-skip-repeating-val` config which is enabled by default.

This config changes the `vis` command rendering so that it collapses a
heap chunk display like this, instead of repeating its lines:
```
  0x555555559000    0x0000000000000000    0x0000000000000291    ................
  0x555555559010    0x0000000000000000    0x0000000000000000    ................
      ... ↓      39 repeated lines skipped
  0x555555559290    0x0000000000000000    0x0000000000000111    ................
```

disconnect3d commented:
This overrides https://github.com/pwndbg/pwndbg/pull/3230

disconnect3d committed:
fix docs

disconnect3d commented:
Here is how it looks now:
<img width="749" height="496" alt="image" src="https://github.com/user-attachments/assets/deab9716-29b7-4de2-b81c-3cdadb21a8ef" />

With skip disabled:
<img width="595" height="877" alt="image" src="https://github.com/user-attachments/assets/f693e295-00fc-4ba2-8b0f-4ff6123f5c70" />

k4lizen commented:
i haven't studied the code but just a few mentions:

1. Should the `-n` flag disable this? I think its probably fine not to
2. Regardless of chunk boundaries, if you encounter an address which is in a freelist or top you should not skip this line
3. There seems to be a bug in `vis` currently where the first line of output is empty, it would be nice if you could drive-by fix this

disconnect3d committed:
Remove empty line at begining of vis command

disconnect3d commented:
1. I added a `--no-skip | -s` flag for this
2. I believe the `if not lables` check covers it
3. I fixed it

I also changed the flags to use `-` instead of `_`.

disconnect3d committed:
vis: add --no-skip|-s flag and use - instead of _ in flags

disconnect3d committed:
fix vis docs

disconnect3d committed:
fix mypy

disconnect3d committed:
fix strict mypy
